### PR TITLE
Use bundled JDK in Sys V init

### LINKED ISSUE #45542: Elasticsearch 7.2.* is not detecting its own JVM java binary on CentOS 6.* or Ubuntu Trusty 14.04
**Elasticsearch version**:
``` 
7.2.0
```

**JVM version**:
```
The issue is related to Java. No Java is previously installed.
```

**OS version**:
```
CentOS 6.* or Ubuntu Trusty 14.04
```

**Description of the problem including expected versus actual behavior**:

Elasticsearch 7.2.* is not detecting its own JVM when starting the service.

**Steps to reproduce**:

Once `Elasticsearch` is installed, then when trying to start it:
```
service elasticsearch start
```

Elasticsearch 7.2.* is not detecting its own JVM when starting the service, implying to get an error message:

```
which: no java in (/sbin:/usr/sbin:/bin:/usr/bin)
Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME
```

**Solution** 

As `Elasticsearch` says in the official documentation, there is no need to have JVM installed in the machine where running `Elasticsearch` because it has an already installed JVM in `Elasticsearch` home folder. In this case the home folder is

```
/usr/share/elasticsearch/jdk/
```

We have to create a symbolic link  to `Java` binary as follows:
```
ln -s /usr/share/elasticsearch/jdk/bin/java /usr/bin/java 
```

Then we can detect the changes by checking `Java` version:
```
java --version
```
```
openjdk 12.0.1 2019-04-16
OpenJDK Runtime Environment (build 12.0.1+12)
OpenJDK 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)
```

**Provide logs (if relevant)**:
`Not Needed in this case`.

**Referecnes**

`Official Docs`: https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html

### PR DESCRIPTION
**Pull request by jasontedor**
This commit addresses an issue when trying to using Elasticsearch on systems with Sys V init and the bundled JDK was not being used. Instead, we were still inadvertently trying to fallback on the path. This commit removes that fallback as that is against our intentions for 7.x where we only support the bundled JDK or an explicit JDK via JAVA_HOME.

Closes #45542

### CODE REVIEW

File: qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTestCase.java [hunk 1]
```diff
@@ -136,6 +136,22 @@ public void test32JavaHomeOverride() throws Exception {
         assertRunsWithJavaHome();
     }
 
+    public void test33RunsIfJavaNotOnPath() throws Exception {
+        assumeThat(distribution().hasJdk, is(true));
+
+        final Result readlink = sh.run("readlink /usr/bin/java");
+        assertThat(readlink.exitCode, equalTo(0));
```
Comment (pugnascotia) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTestCase.java]: `Shell.run` already checks that `exitCode` is 0.

Comment (jasontedor) [reply qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTestCase.java]: Thanks, I pushed 4581efed952c9d6dd9712f5c12589da4812d925d.

**Review by pugnascotia — APPROVED** [review]

File: distribution/packages/src/deb/init.d/elasticsearch [hunk 2]
```diff
@@ -81,16 +81,16 @@ if [ ! -x "$DAEMON" ]; then
...
-		JAVA="$JAVA_HOME/bin/java"
-	else
-		JAVA=`which java`
-	fi
-
-	if [ ! -x "$JAVA" ]; then
-		echo "Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME"
-		exit 1
-	fi
+  if [ ! -z "${JAVA_HOME}" ]; then
```
Comment (rjernst) [inline distribution/packages/src/deb/init.d/elasticsearch]: We don't seem to actually use JAVA here anywhere. Other than exporting JAVA_HOME, could we remove this logic and let bin/elasticsearch handle it?

Comment (jasontedor) [reply distribution/packages/src/deb/init.d/elasticsearch]: I'll explore the consequences of this in a follow-up, I want to get this fix in for 7.3.1.

File: distribution/packages/src/deb/init.d/elasticsearch [hunk 3]
```diff
@@ -81,16 +81,16 @@ if [ ! -x "$DAEMON" ]; then
...
+    JAVA="${JAVA_HOME}"/bin/java
+  else
+    JAVA="${ES_HOME}"/jdk/bin/java
+  fi
+
+  if [ ! -x "$JAVA" ]; then
+    echo "Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME"
```
Comment (rjernst) [inline distribution/packages/src/deb/init.d/elasticsearch]: This error message is still referencing PATH

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**